### PR TITLE
Add tests for documentation

### PR DESCRIPTION
**Pull request by ogenstad**
This is for the 2017 Hackathon
https://github.com/napalm-automation/hackathon-2017/issues/10

This commit validates that each Ansible module has a DOCUMENTATION
and an EXAMPLES variable. It also asserts that those variables are
properly formatted YAML strings.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[tox]
+envlist = py27
+
+[testenv]
+deps = 
+    -rrequirements.txt
+    pytest
+    ansible
```
Comment (ktbyers) [inline tox.ini]: We should create a `requirements-dev.txt` file and have pytest, ansible, tox in that as well as `-r requirements.txt.

**Review by ktbyers — COMMENTED** [review]
I think this is a good idea. 

The only thing I don't like is having two different set of tests defined in two different locations (one in .travis.yml and one in tox.ini).

I am sure I am going to get confused by that in the future.

What extra value is tox providing here (since we are PY27 only)?

**Comment by ogenstad** [comment]
I agree with the comment about duplicate ways to test with tox. Mostly for now that was to give me an easy way to test. The only current value tox brings is that it tests that the package is actually installable and working. But I would say that tox either gets removed later, or the other tests gets moved in there too.

**Comment by dbarrosop** [comment]
Yeah, this is cool. We should consolidate tests to use one tool though as kirk suggested

**Comment by ogenstad** [comment]
I removed tox. Was just a bit lazy and installed a new environment instead. The above might need some other tests as there are some parts of the documentation missing in the modules now (such as `defaults`). Other than that I think this is mostly complete.

Please also comment on https://github.com/napalm-automation/napalm/pull/535 :)

**Comment by ogenstad** [comment]
I think I'm happy with this one now. There are other tests which can be added in the future and some extra controls.

Once this is merged I can add the git clone script in the NAPALM repo which uses the generated json files from here to build the documentation.

**Comment by ogenstad** [comment]
Ok, so now this passes the test. I was only looking at the red paint which should actually be there but missed the:

`The command "./test_changelog.sh" exited with 1.`

Not sure that one always makes sense. Anyway, feel free to review. :)

**Comment by dbarrosop** [comment]
awesome work!

**Pull request merged**